### DISCUSSION TITLE
How to get Node contents

### DISCUSSION THREAD

**Discussion by natanfelles** [thephpleague/commonmark#995] [Q&A] [answered]
Hello!

I need to get the content of paragraphs to add classes to them.

This is my basic configuration:

```php
'default_attributes' => [
    Paragraph::class => [
        'class' => static function (Paragraph $node) {
            $content = $node->getContent(); // Method does not exist
            if (str_starts_with($content, 'danger:')) {
                return 'alert alert-danger';
            }
            return null;
        },
    ],
]
```

Thank you very much!

**Comment by colinodell** [comment, marked as answer]
Hi there!

TL;DR: Use `$content = StringContainerHelper::getChildText($paragraph);`

Paragraph nodes do not directly contain text, but some of their descendants do. Here's an example of such an AST (visualized as XML):

```xml
<document>
  <heading level="1">
    <text>Hello World!</text>
  </heading>
  <paragraph>
    <text>This is an example of </text>
    <strong>
      <text>CommonMark</text>
    </strong>
    <text>.</text>
  </paragraph>
</document>
```

To find that text, you'd need to [iterate through those nodes](https://commonmark.thephpleague.com/2.4/customization/abstract-syntax-tree/), looking for any that implement `StringContainerInterface`, and concatenating what those contain.

Or you can use [this helper function](https://github.com/thephpleague/commonmark/blob/2.4/src/Node/StringContainerHelper.php) which does exactly that for you :)

**Reply by natanfelles** [reply]
Thank you @colinodell !

The following snippet works for me:

```php
'default_attributes' => [
    Paragraph::class => [
        'class' => static function (Paragraph $node) {
            foreach ($node->iterator() as $child) {
                if ($child instanceof StringContainerInterface) {
                    $text = $child->getLiteral();
                    if (str_starts_with($text, 'danger:')) {
                        $child->setLiteral(trim(substr($text, 7)));
                        return 'alert alert-danger';
                    }
                    return null;
                }
            }
        },
    ],
]
```
